### PR TITLE
Initial caching structure documentation

### PR DESCRIPTION
**Pull request by butler54**
Signed-off-by: Chris Butler <chris@thebutlers.me>

## Types of changes

- \[ \] Bug fix (non-breaking change which fixes an issue)
- \[ \] New feature (non-breaking change which adds functionality)
- \[ \] Breaking change (fix or feature that would cause existing functionality to change)
- \[ \] My code follows the code style of this project.
- \[x\] My change requires a change to the documentation.
- \[x\] I have updated the documentation accordingly.
- \[ \] I have added tests to cover my changes.
- \[ \] All new and existing tests passed.

This PR is documenting the caching strategy for upstream references to OSCAL documents. It is currently work in progress. I will keep adding detail and then ask for an initial review.

### CODE REVIEW

File: docs/caching_spec.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+- [Git support](<>)
+- [Update behaviour](<>)
+
+## Purpose
+
+This document describes the caching strategy and behaviour that is supported by trestle for 3rd party artifacts that are
+external to the core repo. A principle that trestle is expected to uphold is that documents, unless the user explicitly
+contradicts, are referred to by reference.
+
+For example the user should *not* be expected to manually copy NIST 800-53 into their working directory (although they may). They waould refer to the source document such that integrity is maintained. Given this a caching strategy is required to manage these documents - especially when operating in a CICD world.
```
Comment (brunomarq) [inline docs/caching_spec.md:19]: *waould = would

File: docs/caching_spec.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+## Scope
+
+This document is only concerned with links which refer to other OSCAL schema objects. External references (say to an xccdf document) are not considered.
+
+## Types of external references.
+
+NIST defines many types of referential links within the OSCAL schema. Here trestle is interested in supporting a subset of these links. In particular these links are characterized specifically by the fact that they refer to 'upstream' OSCAL
+artifacts. The table below identifies the references covered by this document.
+
+| Schema               | Json path (\[*\] represents array or named keys)                      | Refers to            |
```
Comment (brunomarq) [inline docs/caching_spec.md:34]: Should we also include `target-definition` in this table?

File: docs/caching_spec.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+| \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-| \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-- | \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\--- |
+| profile              | profile.imports\[*\].href                                             | catalog or profile   |
```
Comment (brunomarq) [inline docs/caching_spec.md:36]: Should we include `profile.merge` and `profile.modify`?

**Review by brunomarq — CHANGES_REQUESTED** [review]

File: docs/caching_spec.md (hunk 3 above)
Comment (butler54) [reply docs/caching_spec.md:36]: profile.merge and profile.modify do not have external hrefs. Controls must first be imported.

**Review by brunomarq — APPROVED** [review]
LGTM

**Pull request merged**